kernel/src/misc/wm/wm.c

```diff
@@ -489,65 +509,55 @@ void wm_mouse_callback(mouse_t raw_curr) {
     mouse.x = mouse.x < 0 ? 0 : mouse.x;
     mouse.y = mouse.y < 0 ? 0 : mouse.y;
 
-    // Raises and drag starts
+    // Is the button down?
     if (!raw_prev.left_pressed && raw_curr.left_pressed) {
-        wm_window_t* clicked = wm_window_at(mouse.x, mouse.y);
+        lmb_is_down = true;
+    } else if (!raw_curr.left_pressed) {
+        lmb_is_down = false;
+    }
 
-        if (clicked) {
-            wm_raise_window(clicked);
-            dragged = clicked;
+    // Raise window
+    if (lmb_is_down){
+        if ( (win = wm_window_at(mouse.x, mouse.y)) ) {
+            wm_raise_window(win);
         }
     }
 
-    // Drags
-    if (raw_prev.left_pressed && raw_curr.left_pressed) {
-        if (dragged && (dx || dy)) {
-            rect_t rect = rect_from_window(dragged);
+    // Drag window
+    if (lmb_is_down) {
+        // only drag window when hovering title bar and 
+        if ((dx || dy) && wm_hovering_tb_at(mouse.x, mouse.y)) {
+            rect_t rect = rect_from_window(win);
 
-            if (!(rect.left + dx < 0 || rect.right + dx >= (int32_t) fb.width ||
-                    rect.top + dy < 0 || rect.bottom + dy >= (int32_t) fb.height)) {
-                been_dragged = true;
+            win->x += dx;
+            win->y += dy;
```

Comment:
The issue with the window moving slower than the cursor when dragging is related to these two lines:
- the cursor movements dx/dy are floats
- window coordinates are integers
- the decimal part of dx and dy are lost at every step of a drag motion

Two ways to fix this I can think of:
- make window coordinates floating points. This might be a source of headaches as it touches a lot of stuff.
- probably better, keep a cumulative dx/dy by having them static and incremented by the mouse movement each call. Also track the initial position of the dragged window at the start of the event, and recalculate its position with something like `initial_pos.x/y + cumulative_dx/dy`.

Actually I don't really like the first idea, second one is much easier and to the point.